shell/platform/windows/settings_plugin.cc

```diff
@@ -103,19 +103,29 @@ float SettingsPlugin::GetTextScaleFactor() {
 }
 
 SettingsPlugin::PlatformBrightness SettingsPlugin::GetPreferredBrightness() {
-  DWORD use_light_theme;
-  DWORD use_light_theme_size = sizeof(use_light_theme);
-  LONG result = RegGetValue(HKEY_CURRENT_USER, kGetPreferredBrightnessRegKey,
-                            kGetPreferredBrightnessRegValue, RRF_RT_REG_DWORD,
-                            nullptr, &use_light_theme, &use_light_theme_size);
-
-  if (result == 0) {
-    return use_light_theme ? SettingsPlugin::PlatformBrightness::kLight
-                           : SettingsPlugin::PlatformBrightness::kDark;
+  if (is_high_contrast_) {
+    DWORD text_color = GetSysColor(COLOR_WINDOW);
+    int r = GetRValue(text_color);
+    int g = GetGValue(text_color);
+    int b = GetBValue(text_color);
+    int luminance = (r + r + r + b + (g << 2)) >> 3;
+    return luminance >= 0.5 ? SettingsPlugin::PlatformBrightness::kLight
+                            : SettingsPlugin::PlatformBrightness::kDark;
   } else {
-    // The current OS does not support dark mode. (Older Windows 10 or before
-    // Windows 10)
-    return SettingsPlugin::PlatformBrightness::kLight;
+    DWORD use_light_theme;
+    DWORD use_light_theme_size = sizeof(use_light_theme);
+    LONG result = RegGetValue(HKEY_CURRENT_USER, kGetPreferredBrightnessRegKey,
+                              kGetPreferredBrightnessRegValue, RRF_RT_REG_DWORD,
+                              nullptr, &use_light_theme, &use_light_theme_size);
```

Comment:
Similarly here, it might be useful to extract this to a named function with a short doc comment.